### ISSUE TITLE
ESLint Config inspector fails to load with `eslint-plugin-mdx`

### ISSUE THREAD

**Issue by Simek** [mdx-js/eslint-mdx#598]
### Initial checklist

- [x] I read the [support docs](https://mdxjs.com/community/support/)
- [x] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [x] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/main/code-of-conduct.md)
- [x] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=issues) and [discussions](https://github.com/orgs/mdx-js/discussions) and couldn’t find anything (or linked relevant results below)

### Affected package

eslint-plugin-mdx

### Steps to reproduce

1. Check out https://github.com/facebook/react-native-website.
2. Run  `npx eslint --inspect-config` at the root to see the failure.
3. Comment out `eslint-plugin-mdx` import and its usages in `eslint.config.js`:
   ```tsx
   import * as eslintPluginMdx from 'eslint-plugin-mdx';

   [...]

   ...eslintPluginMdx.flat,
   processor: eslintPluginMdx.createRemarkProcessor({
     lintCodeBlocks: false,
     remarkConfigPath: 'website/.remarkrc.mjs',
   }),
   ```
4. Re-run  `npx eslint --inspect-config`, it should work now.

Refs:
* https://github.com/eslint/config-inspector/issues/201

### Actual behavior

ESLint Inspect Config tool fails to load the config which use `eslint-plugin-mdx`:

```
Failed to load eslint.config.js
Error: Could not find ESLint Linter in require cache
    at file:///W:/GitHub/react-native-website/node_modules/eslint-plugin-mdx/lib/processors/options.js:5:11
    at ModuleJob.run (node:internal/modules/esm/module_job:377:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:671:26)
    at async extractResult (file:///C:/Users/Simek/AppData/Local/npm-cache/_npx/fadb285f08a830ab/node_modules/bundle-require/dist/index.js:169:15)
    at async run (file:///C:/Users/Simek/AppData/Local/npm-cache/_npx/fadb285f08a830ab/node_modules/bundle-require/dist/index.js:211:17)
```

I look like the require cache check is a the root of the issue:
* https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-plugin-mdx/src/processors/options.ts#L14-L22

### Expected behavior

ESLint Inspect Config should be able to load the config, and start correctly.

### Runtime

node@22.21.1 and node@24.11.1

### Package manager

yarn@4.12.0

### Operating system

OS agnostic (tested on macOS and Windows)

### Build and bundle tools

Docusaurus
